source/ceylon/time/iso8601/parseTimeZone.ceylon

```diff
@@ -1,200 +1,66 @@
-import ceylon.time.base { ms = milliseconds }
 import ceylon.time.timezone {
-    timeZone,
-    TimeZone
+    TimeZone,
+    OffsetTimeZone
+}
+import ceylon.time.base {
+    milliseconds
 }
-
-
 
 "Timezone offset parser based on ISO-8601, currently it accepts the following 
  time zone offset patterns:
  
- - &plusmn;`[hh]:[mm]`, 
+ - &plusmn;`[hh]:[mm]`,
  - &plusmn;`[hh][mm]`, and 
  - &plusmn;`[hh]`.
  
  In addition, the special code `Z` is recognized as a shorthand for `+00:00`"
-shared TimeZone|ParserError parseTimeZone( String offset ) {
-    variable TzState state = TzInitial();
-    for( character in offset ) {
-        state = state.next( Chunk( character ) );
+shared TimeZone? parseTimeZone( String offset ) {
+    value signal = offset.startsWith("+") || offset.equalsIgnoringCase("Z")
+            then 1 
+            else (offset.startsWith("-") 
+                then -1
+                else null);
+    value offsetWithoutSignal = offset.spanFrom(1);
+    Integer? hours;
```

Comment:
wouldnät it be easier to check the length of the time zone string first and parse fixed ranges based on the length (there are only 3 legal lengths for tz qualifier string: 2, 4 and 5)